lib/complete-provider.coffee

```diff
@@ -53,11 +58,19 @@ class CompleteProvider extends Provider
     'then', 'where'
   ]
 
-  initialize: (@editorView, @manager) ->
+  constructor: (@editor, @manager) ->
+    if (@editor.haskellCompletionProvider && @editor.haskellCompletionProvider != this)
+      # shouldn't happen, but log just in case
+      console.log "weird: editor already has a haskell completion provider"
+      @editor.haskellCompletionProvider.dispose()
```

Comment:
FuzzyProvider seemingly rebuilds word list every time pane item is changed, does it not? While that's OK for simple cases, running ghc-mod is actually quite expensive both in terms of memory and cpu time. So I'd prefer having a caching provider instance for each editor rather than re-run ghc-mod every time. Alternative would be to store *cache* in a WeakMap inside (singleton) provider instance. That's also acceptable. But it's basically the same thing, and I think requires a little more boilerplate (but I might be wrong)